packages/trie/src/types.ts

```diff
@@ -8,9 +8,18 @@ export type TrieNode = BranchNode | ExtensionNode | LeafNode
 
 export type Nibbles = number[]
 
+// A raw node refers to the non-serialized, array form of the node
+// A raw extension node is a 2-item node, where the first item is the encoded path to the next node, and the second item is the reference to the next node
+// A raw leaf node is a 2-item node, where the first item is the remaining path to the leaf node, and the second item is the value
+// To learn more: https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/#optimization
```

Comment:
Yes, great that the extra docs on top have been added. In the future this external URL could 404, so in general we should add extra comments if we point to external URLs :smile: :+1: